### ISSUE TITLE
[Accessibility]: Keyboard trap in optgroup variant of combo box

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/vets-design-system-documentation#5113]
### Point of contact

Jeana Clark

### Defect Severity

1, Launchblocker. Critical. Must be fixed before launch.

### Severity rationale

When someone interacts with the combo box that has an opt group, some searches could result in a keyboard trap. If the user tabs away they can get out of the combo box, but they can't continue with the same expected keyboard navigation in certain circumstances.

### VA 508 Office

N/A

### URL where issue occurs

https://design.va.gov/storybook/iframe.html?globals=&args=&id=uswds-va-combo-box--option-groups&viewMode=story

### Component or pattern (if applicable)

`va-combo-box` with optgroups

### Environment details

Tested on
- Mac Chrome, VO
- Mac Safari, VO
- Windows Edge, JAWS
- Windows Chrome, JAWS
- Windows Firefox, JAWS
- Windows Edge, NVDA
- Windows Chrome, NVDA
- Windows Firefox, NVDA
- iOS Safari, VO
- iOS Chrome, VO
    - In iOS I could navigate the combo box okay and not get trapped, however the `<opt>` group does become selectable and I don't think that's expected. 

### Steps to reproduce

1. Go to https://design.va.gov/storybook/iframe.html?globals=&args=&id=uswds-va-combo-box--option-groups&viewMode=story
2. Type "e" (this at least gets the Optgroup Vegetable selected and other items)
3. Vegetable as an opt group displays twice.
4. When you arrow down to one of them, keyboard gets trapped.

### Observed behavior

The keyboard gets trapped using the expected arrow keys and you can't get past the duplicated `<opt>` group.

### Expected behavior / Acceptance criteria

1. Opt groups shouldn't be duplicated
2. I should be able to navigate through all items in the combo box with the keyboard arrow key (opt group labels should be skipped as they are not selectable)

### Supporting evidence (screenshots, recordings, code snippets)

This is a video of it happening in form 21-4140

https://github.com/user-attachments/assets/2791fb5e-f4bb-4a92-92ca-10ad711e7ef5

### Related issues or systemic notes

This violates [WCAG 2.1.2](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap.html) - No Keyboard trap

### Type of issue

- [ ] Axe-core
- [ ] Cognitive | user flows & navigating
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [ ] Focus
- [ ] Headings
- [x] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Screenreader | assistive tech & device support
- [ ] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else
